app/src/main/java/com/spendster/presentation/addExpenses/AddExpensesPresenter.java

```diff
@@ -1,48 +1,48 @@
 package com.spendster.presentation.addExpenses;
 
-import android.content.Context;
 import android.content.res.Resources;
 
 import com.spendster.R;
 import com.spendster.data.entity.Expense;
-import com.spendster.presentation.authentication.SharedPreferencesUserStorage;
+import com.spendster.presentation.authentication.SUserStorage;
+import com.spendster.presentation.utils.TextDate;
 
 import java.text.DecimalFormat;
+import java.text.ParseException;
+import java.util.Date;
 
 import io.reactivex.android.schedulers.AndroidSchedulers;
 import io.reactivex.disposables.CompositeDisposable;
 import io.reactivex.observers.DisposableSingleObserver;
 import io.reactivex.schedulers.Schedulers;
 
 public class AddExpensesPresenter {
-    private final Context context;
+    private final SUserStorage sUserStorage;
     private final AddExpensesView addExpensesView;
     private final ServerAddExpensesModel serverAddExpensesModel;
     private final CompositeDisposable compositeDisposable = new CompositeDisposable();
 
     public AddExpensesPresenter(AddExpensesView addExpensesView, ServerAddExpensesModel
-            serverAddExpensesModel, Context contex) {
+            serverAddExpensesModel, SUserStorage sUserStorage) {
         this.addExpensesView = addExpensesView;
         this.serverAddExpensesModel = serverAddExpensesModel;
-        this.context = contex;
+        this.sUserStorage = sUserStorage;
     }
 
-    public void save(double amount, String title, String note, String categoryId, long date) {
-        // validation
-        SharedPreferencesUserStorage sharedPreferencesUserStorage = new SharedPreferencesUserStorage(context);
-        String userId = sharedPreferencesUserStorage.read().getUserId();
+    public void save(double amount, String title, String note, String categoryId, TextDate textDate) {// validation
+        long date = new Date().getTime();
+        try {
+            date = textDate.date().getTime();
+        } catch (ParseException pe) {
+            this.addExpensesView.showError("Bad date format");
+        }
+        String userId = this.sUserStorage.read().getUserId();
         DecimalFormat decimalFormat = new DecimalFormat("#.##");
         amount = Double.valueOf(decimalFormat.format(amount));
-        if (title.equals(Resources.getSystem().getString(R.string.expenses)) && categoryId.isEmpty()) {
-            addExpensesView.showError("Category is not selected");
-        }
-        if (note.isEmpty()) {
-            addExpensesView.showError("Write some notes");
-        }
-        // request
+        validation(title, note, categoryId);
```

Comment:
@SimonMaha what if validation did not pass?

Reply:
We receive an error from validation method